llm-complete-guide/steps/populate_index.py

```diff
@@ -626,36 +632,35 @@ def generate_embeddings(
 class IndexType(Enum):
     ELASTICSEARCH = "elasticsearch"
     POSTGRES = "postgres"
+    PINECONE = "pinecone"
 
 
 @step(enable_cache=False)
 def index_generator(
     documents: str,
-    index_type: IndexType = IndexType.POSTGRES,
+    index_type: IndexType = IndexType.PINECONE,
 ) -> None:
     """Generates an index for the given documents.
 
-    This function creates a database connection, installs the pgvector extension if not already installed,
-    creates an embeddings table if it doesn't exist, and inserts the embeddings and document metadata into the table.
-    It then calculates the index parameters according to best practices and creates an index on the embeddings
-    using the cosine distance measure.
-
     Args:
-        documents (str): A JSON string containing the Document objects with generated embeddings.
-        index_type (IndexType): The type of index to use. Defaults to Elasticsearch.
-
-    Raises:
-        Exception: If an error occurs during the index generation.
+        documents (str): JSON string containing the documents to index.
+        index_type (IndexType, optional): Type of index to generate. Defaults to IndexType.POSTGRES.
     """
-    try:
-        if index_type == IndexType.ELASTICSEARCH:
-            _index_generator_elastic(documents)
-        else:
-            _index_generator_postgres(documents)
+    # get model version
+    context = get_step_context()
+    model_version_name_or_id = context.model_version.name
```

Comment:
what behavior are you trying to achieve here with the model version?

Reply:
This is the implementation of the 'separate vector DBs for staging and prod environments thing that you requested a while back. It's not a perfect solution, but I'll let @wjayesh explain the workflow as he was the one who worked on this part.

Reply:
The idea is that for any version that is not in production, we use a different vector DB. And for production, we have a different one that is built from scratch when the interaction pipeline runs with the production version

Reply:
I assumed in order to get a specific "version" of the embedding db, you would need to go through our model control plane to find the link to the embeddings db at a certain version